File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/AdditionalConsentModeDataSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.AdTechProvider
import com.usercentrics.sdk.AdditionalConsentModeData

internal fun AdditionalConsentModeData.serialize(): Any {
    return mapOf(
        "acString" to acString,
        "adTechProviders" to adTechProviders.map { it.serialize() }
    )
}

private fun AdTechProvider.serialize() : Any {
    return mapOf(
        "id" to id,
        "name" to name,
        "privacyPolicyUrl" to privacyPolicyUrl,
        "consent" to consent
    )
}

File: ios/Classes/Serializer/AdditionalConsentModeDataSerializer.swift
import Usercentrics

extension AdditionalConsentModeData {
    func serialize()-> Any {
        return [
            "acString" : acString,
            "adTechProviders" : adTechProviders.map { $0.serialize() }
        ] as [String : Any]
    }
}

extension AdTechProvider {
    func serialize()-> Any {
        return [
            "id" : id,
            "name" : name,
            "privacyPolicyUrl" : privacyPolicyUrl,
            "consent" : consent
        ] as [String : Any]
    }
}
